### ISSUE TITLE
IPTV Merge: 0.24.8 regression - sat channels epg broken

### ISSUE THREAD

**Issue by Paul608085** [matthuisman/slyguy.addons#637]
I have had an issue recently with a dual appearance of a single channel in my IPTV playlist appearing twice in the EPG, on Kodi v20 only - see closed Issue 636. @matthuisman provided an update yesterday for IPTV Merge to v0.24.8

I think that this update has had an unintended consequence - at least for me:

On my Kodi installations I also use the TV-Headend plugin to add British free satellite channels. These appear on the EPG as genres "Freesat" and "Regional" and sit neatly at the end of the top row, after all of the IPTV genres listed in my playlist. The satellite channels get their EPG data "over-the-air".

Since Merge's upgrade yesterday, The satellite channels have not been updating EPG data, but the IPTV channels are still fine. I have tried clearing the Guide data in both locations in Kodi. Although it usually takes a few minutes for the EPG data to repopulate, (sometimes up to 10 minutes for all of the satellite channels), I still have no EPG data for the satellite channels after 2 hours. The satellite channels can still be viewed, just no EPG.

Again, this issue only affects my Firestick with the Kodi v20 installation, both devices using Kodi v19 are still working fine, even though they have also upgraded to Merge v0.24.8

Am I correct in linking this to the Merge update - or is it just a co-incidence?

Whereas the previous issue that the update solved was annoying, no EPG data for any of my satellite channels is far more problematic.

If the "solution" is to revert to Kodi v19.5 on this Firestick, then I'm happy to do that - I wanted to test v20 again as I'm concerned about my installations becoming out of date and unsupported in the future?

EDIT: If I disable IPTV Merge and IPTV Simple Client, the satellite channels EPG populates correctly.

**Comment by matthuisman** [comment]
can you download https://slyguy.uk/.repo/repository.slyguy/plugin.program.iptv.merge/plugin.program.iptv.merge-0.24.7.zip and install it and see if it fixes the issue?

It was a pretty small change yesterday but ya never know

**Comment by Paul608085** [comment]
Hi @matthuisman 

I have now uninstalled Merge v0.24.8, (including its data), and re-installed v0.24.7 from your link.

Both IPTV and satellite channels have EPG data again - along with the original issue of the first channel in the playlist appearing incorrectly at the top of the last genre, (as expected). 

Building on your suggestion in the Issue 636 thread, that an empty group might resolve the dual appearance issue, I have just been experimenting with adding an extra semi-colon at the end of the #EXTGRP line in my playlist - both with and without a space, (ie ;; and ; ;), but all this seemed to do was prevent the IPTV channels from loading at all!

TBH, the original issue that I reported, the dual appearance of a single channel, was no more than an observation that I'm happy to ignore if it is difficult to resolve? Equally, I'm happy to co-operate with testing to correct it.
